packages/web/components/one-click-trading/screens/reset-period-screen.tsx

```diff
@@ -0,0 +1,80 @@
+import { OneClickTradingResetPeriods } from "@osmosis-labs/types";
+import classNames from "classnames";
+
+import { Button } from "~/components/buttons";
+import { OneClickTradingBaseScreenProps } from "~/components/one-click-trading/screens/types";
+import { ScreenGoBackButton } from "~/components/screen-manager";
+import { useTranslation } from "~/hooks";
+
+const ResetPeriods = [
+  "day",
+  "week",
+  "month",
+  "year",
+] as OneClickTradingResetPeriods[];
+
+export function getResetPeriodTranslationKey(
+  id: OneClickTradingResetPeriods
+): string {
+  switch (id) {
```

Comment:
If the key is always in the same format as the corresponding translation, couldn't we just check if "ResetPeriods" contains that key and then construct the string as "oneClickTrading.settings.resetPeriodScreen.periods.{key}"? Obviously if it doesn't at contains we return "oneClickTrading.settings.resetPeriodScreen.periods.unknown", so in the future if we add other keys we only need to update "ResetPeriods" and the translations and we won't have a long switch.

What do you think?

Reply:
This is a good idea! I'll add this in